bin/runtime-common/src/refund_relayer_extension.rs

```diff
@@ -123,71 +221,30 @@ where
 	}
 }
 
-/// Data that is crafted in `pre_dispatch` method and used at `post_dispatch`.
-#[derive(PartialEq)]
-#[cfg_attr(test, derive(Debug))]
-pub struct PreDispatchData<AccountId> {
-	/// Transaction submitter (relayer) account.
-	relayer: AccountId,
-	/// Type of the call.
-	pub call_type: CallType,
-}
-
-/// Type of the call that the extension recognizes.
-#[derive(Clone, Copy, PartialEq, RuntimeDebugNoBound)]
-pub enum CallType {
-	/// Relay chain finality + parachain finality + message delivery calls.
-	AllFinalityAndDelivery(RelayBlockNumber, SubmitParachainHeadsInfo, ReceiveMessagesProofInfo),
-	/// Parachain finality + message delivery calls.
-	ParachainFinalityAndDelivery(SubmitParachainHeadsInfo, ReceiveMessagesProofInfo),
-	/// Standalone message delivery call.
-	Delivery(ReceiveMessagesProofInfo),
-}
-
-impl CallType {
-	/// Returns the pre-dispatch messages pallet state.
-	fn receive_messages_proof_info(&self) -> ReceiveMessagesProofInfo {
-		match *self {
-			Self::AllFinalityAndDelivery(_, _, info) => info,
-			Self::ParachainFinalityAndDelivery(_, info) => info,
-			Self::Delivery(info) => info,
-		}
-	}
-}
-
-// without this typedef rustfmt fails with internal err
-type BalanceOf<R> =
-	<<R as TransactionPaymentConfig>::OnChargeTransaction as OnChargeTransaction<R>>::Balance;
-type CallOf<R> = <R as frame_system::Config>::RuntimeCall;
-
-impl<R, GI, PI, MI, PID, LID, FEE> SignedExtension
-	for RefundRelayerForMessagesFromParachain<R, GI, PI, MI, PID, LID, FEE>
+impl<Runtime, Para, Msgs, Refund, Id> SignedExtension
+	for RefundBridgedParachainMessages<Runtime, Para, Msgs, Refund, Id>
 where
-	R: 'static
-		+ Send
-		+ Sync
-		+ UtilityConfig<RuntimeCall = CallOf<R>>
-		+ BoundedBridgeGrandpaConfig<GI>
-		+ ParachainsConfig<PI, BridgesGrandpaPalletInstance = GI>
-		+ MessagesConfig<MI>
+	Self: 'static + Send + Sync,
+	Runtime: UtilityConfig<RuntimeCall = CallOf<Runtime>>
+		+ BoundedBridgeGrandpaConfig<Runtime::BridgesGrandpaPalletInstance>
+		+ ParachainsConfig<Para::Instance>
+		+ MessagesConfig<Msgs::Instance>
 		+ RelayersConfig,
-	GI: 'static + Send + Sync,
-	PI: 'static + Send + Sync,
-	MI: 'static + Send + Sync,
-	PID: 'static + Send + Sync + Get<u32>,
-	LID: 'static + Send + Sync + Get<LaneId>,
-	FEE: 'static + Send + Sync + TransactionFeeCalculation<<R as RelayersConfig>::Reward>,
-	CallOf<R>: Dispatchable<Info = DispatchInfo, PostInfo = PostDispatchInfo>
-		+ IsSubType<CallableCallFor<UtilityPallet<R>, R>>
-		+ GrandpaCallSubType<R, GI>
-		+ ParachainsCallSubType<R, PI>
-		+ MessagesCallSubType<R, MI>,
+	Para: RefundableParachainId,
+	Msgs: RefundableMessagesLaneId,
+	Refund: RefundCalculator<Balance = Runtime::Reward>,
+	Id: StaticStrProvider,
```

Comment:
@serban300 
can we use here just `Get<&'static str>` instead of StaticStrProvider?

Reply:
No, it would have been nice, but it doesn't work :( Because Get<...>::get is not `const`.

Reply:
I see one usage: `N: Get<&'static str>`  for some staking migrations:
```
pub struct StakingMigrationV11OldPallet;
impl Get<&'static str> for StakingMigrationV11OldPallet {
	fn get() -> &'static str {
		"VoterList"
	}
}
```

Reply:
Yes, tried again. I'm getting:

```
cannot call non-const fn `<Id as frame_support::traits::Get<&str>>::get` in constants
```

We'll be able to use const generics in the future when they are stable. Until then we need this workaround.